lib/json/api/to_hash.rb

```diff
@@ -0,0 +1,9 @@
+module JSON
+  module API
+    module ToHash
+      def to_hash
```

Comment:
Mixing-in modules is definitely a cool feature of ruby, but in this specific case I don't think it is the right choice (reduces code clarity, might lead to bugs if the hash's name gets changed for some reason).